pkg/select/blockdevice/filters_test.go

```diff
@@ -164,6 +191,42 @@ func TestFilterBlockDeviceTag(t *testing.T) {
 			},
 			wantedNoofBDs: 2,
 		},
+		"some BDs with tag key, but with empty selector": {
+			args: args{
+				bdLabelList: bdLabelList4,
+				spec: &apis.DeviceClaimSpec{
+					Selector: &v1.LabelSelector{},
+				},
+			},
+			wantedNoofBDs: 2,
+		},
+		"some BDs with tag key, with selector matching empty tag": {
+			args: args{
+				bdLabelList: bdLabelList4,
+				spec: &apis.DeviceClaimSpec{
+					Selector: &v1.LabelSelector{
+						MatchLabels: map[string]string{kubernetes.BlockDeviceTagLabel: ""},
+					},
+				},
+			},
+			wantedNoofBDs: 4,
```

Comment:
Also, is it allowed to pass a BD tag with empty value in BDC. 
  
Shouldn't this be 0?  because `""` will always be excluded.

Reply:
cc: @akhilerm

Reply:
> Also, is it allowed to pass a BD tag with empty value in BDC.

Yes. it is allowed, because its a valid label selector.

> Shouldn't this be 0? because "" will always be excluded.

Actually 2 levels of filtering happens. The original BD List that we get here, will be after a label selection as part of the list call. So those selectors will already be applied. But if those selectors are applied, we still need to remove the empty tag, thats being done in this filter.

So the tests assume that already a selection has happened with the list call. And the output of the tests will be therefore including the BDs that would have been removed by the select call. Thats why the additional 4 BDs come.

Reply:
If you check [this](https://github.com/openebs/node-disk-manager/pull/500/files#diff-10b936d2720dd7e71a105e22758229744715db30e38ca881ed9b9f282fc044b7R288) comment , you can see how the filter works.